File: 0815/Java/palindrome.java
import java.util.Scanner;

public class Program {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.println("Enter a string:");
        String input = scanner.nextLine();
        
        boolean result = isPalindrome(input);
        System.out.println("Is the string a palindrome? " + result);
        
        scanner.close();
    }

    static boolean isPalindrome(String str) {
        String clean = str.replaceAll("[\\W_]", "").toLowerCase();
        String reversed = new StringBuilder(clean).reverse().toString();
        return clean.equals(reversed);
    }
}

File: 0815/C#/palindrome.cs
using System;
using System.Text.RegularExpressions;
using System.Linq;

class Program
{
    static void Main()
    {
        Console.WriteLine("Enter a string:");
        string input = Console.ReadLine();
        
        bool result = IsPalindrome(input);
        Console.WriteLine($"Is the string a palindrome? {result}");
    }

    static bool IsPalindrome(string str)
    {

        string clean = new string(str.Where(c => !char.IsPunctuation(c) && !char.IsWhiteSpace(c)).ToArray()).ToLower();

        return clean == new string(clean.Reverse().ToArray());
    }
}